Agda.cabal

```diff
@@ -614,7 +614,7 @@ library
   default-language: Haskell2010
 
   default-extensions: ConstraintKinds
-                    , DataKinds
+--                    , DataKinds
```

Comment:
Please add a comment here why you removed this extension, linking to this PR #3495 .

(Otherwise, in two year's time, someone else will come asking himself "Why don't we have `DataKinds` enabled in `Agda.cabal`, and put it back...) :D